improve github PR template (#3559)
pingcap/tidb-operator: .github/pull_request_template.md

<!--

> Thank you for contributing to TiDB Operator! Please read TiDB Operator's [CONTRIBUTING](https://github.com/pingcap/tidb-operator/blob/master/CONTRIBUTING.md) document **BEFORE** filing this PR.

Reviewer: How about still keep the link to `CONTRIBUTING.md`?

Author: kept

-->

### What is changed and how does it work?
<!--
Please describe the design that your implementation follows AS DETAILED AS POSSIBLE.

For example:

    The root cause is a nil pointer dereferencing. Add a `ptr != nil` check before access members of `ptr` to prevent the crash.
-->

### Code changes

- [ ] Has Go code change
- [ ] Has CI related scripts change

### Tests
<!-- AT LEAST ONE test must be included. -->

- [ ] Unit test <!-- If you added any unit test cases, check this box -->
- [ ] E2E test <!-- If you added any e2e test cases, check this box -->
- [ ] Manual test <!-- If this PR needs manual test, check this box, and add detailed manual test scripts or steps below, so that ANYONE CAN REPRODUCE IT. Ref: https://github.com/pingcap/tidb-operator/pull/3517 -->
- [ ] No code <!-- If this PR contains no code changes, check this box -->

### Side effects

- [ ] Breaking backward compatibility <!-- If this PR breaks things deployed with previous TiDB Operator versions, check this box -->
- [ ] Other side effects: <!-- Any other side effects, such as requiring additional storage / consumes substantial memory / potential reconcilation latency -->

### Related changes

> - [ ] Need to cherry-pick to the release branch <!-- If this PR should also appear in the current release branch, check this box -->

Reviewer: we have two release branches, how about given two checkboxes for both versions?

Author: Then this template must be changed every time we add a new release branch
Is there a better way of doing this? 🤔

Reviewer: I prefer to change this template for every release (we know the PR should be cherry-picked to which branch(es)). but still waiting for opinions from other reviewers.

Author: I don't think change template with code release is a good idea.
Maybe we could make the committer write the release number?
WDYT @lonng @DanielZhangQD

Reviewer 2: I think it's enough to let the author of PR marks whether this PR need to be cherry-picked, and the reviewer add some labels for dedicate release branches.

<!--
Thank you for contributing to TiDB Operator!

> Please complete the following template before creating a PR

Author: ```suggestion
Please complete the following template before creating a PR.
```

> Please complete the following template before creating a PR :D

Author: ```suggestion
Please complete the following template before creating a PR
```

> Please complete the following template before creating a PR :D

Reviewer 3: ```suggestion
Please complete the following template before creating a PR.
```

Ref: TiDB Operator [CONTRIBUTING](https://github.com/pingcap/tidb-operator/blob/master/CONTRIBUTING.md) document
-->

### What problem does this PR solve?
<!--
Please describe the problem AS DETAILED AS POSSIBLE.

> Add and ISSUE LINK WITH SUMMARY if exists.

Reviewer 3: ```suggestion
Add an ISSUE LINK WITH SUMMARY if exists.
```

For example:

    The root cause is a nil pointer dereferencing. Add a `ptr != nil` check before access members of `ptr` to prevent the crash.
-->

### Code changes

- [ ] Has Go code change
- [ ] Has CI related scripts change

### Tests
<!-- AT LEAST ONE test must be included. -->

- [ ] Unit test <!-- If you added any unit test cases, check this box -->
- [ ] E2E test <!-- If you added any e2e test cases, check this box -->
- [ ] Manual test <!-- If this PR needs manual test, check this box, and add detailed manual test scripts or steps below, so that ANYONE CAN REPRODUCE IT. Ref: https://github.com/pingcap/tidb-operator/pull/3517 -->
- [ ] No code <!-- If this PR contains no code changes, check this box -->

### Side effects

- [ ] Breaking backward compatibility <!-- If this PR breaks things deployed with previous TiDB Operator versions, check this box -->
- [ ] Other side effects: <!-- Any other side effects, such as requiring additional storage / consumes substantial memory / potential reconciliation latency -->

### Related changes

- [ ] Need to cherry-pick to the release branch <!-- If this PR should also appear in the current release branch, check this box -->

> - [ ] Need to update the documentation <!-- If this PR introduces new features / change previous usages, check this box -->

Reviewer 3: ```suggestion
- [ ] Need to update the documentation <!-- If this PR introduces new features or changes previous usages, check this box -->
```
